keps/sig-api-machinery/2876-crd-validation-expression-language/README.md

```diff
@@ -515,18 +570,20 @@ The initial changes made in the type integration will be:
   - We couldn't have both: (a) equality of string keys, which implies a canonical ordering of key fields, and (b) ability for developers to successfully lookup a map entry regardless of how they order the keys in the string representation
 
 
-So instead of treating "associative lists" as maps in CEL, we will continue to treat them as lists, but override equality to ignore object order (i.e. use map equality semantics) and introduce utility functions to make the list representation easy to use in CEL. E.g. instead of:
+So instead of treating "associative lists" as maps in CEL, we will continue to treat them as lists, but override equality to ignore object order (i.e. use map equality semantics).
+
+Looking up entiries by keys is available primarily via the `exists_one` and `filter` macros. Examples:
```

Comment:
sounds like `filter` is the example I was looking for above?

Reply:
For the envars case? Here's the ways I can think to do it:

```
!('MY_ENV' in self.envars) || self['MY_ENV'].matches('^[a-zA-Z]*$') // my above example
'MY_ENV' in self.envars ? self['MY_ENV'].matches('^[a-zA-Z]*$') : true
self.envvars.filter(k, k == 'MY_ENV').all(k, self.envvars[k].matches('^[a-zA-Z]*$')
```

Reply:
Added an example for a list map.